### PR TITLE
Create space with zone

### PR DESCRIPTION
**Pull request by darionyaphet**

### CODE REVIEW

File: src/parser/parser.yy [hunk 1]
```diff
@@ -2720,11 +2720,22 @@ create_space_sentence
         auto sentence = new CreateSpaceSentence($4, $3);
         $$ = sentence;
     }
+    | KW_CREATE KW_SPACE opt_if_not_exists name_label KW_ON name_label {
+        auto sentence = new CreateSpaceSentence($4, $3);
+        sentence->setGroupName($6);
+        $$ = sentence;
+    }
```
Comment (bright-starry-sky) [inline src/parser/parser.yy:2727]: A little question， Why not use attribute value？ such as : 
CREATE SPACE default_space(partition_num=9, replica_factor=3, group_name="group0")

Comment (darionyaphet) [reply src/parser/parser.yy:2727]: Maybe it's a different design. In my opinion, group name is not a property for space

Comment (bright-starry-sky) [reply src/parser/parser.yy:2727]: > Maybe it's a different design. In my opinion, group name is not a property for space

Got, nice. LGTM.

File: src/executor/admin/SpaceExecutor.cpp [hunk 2]
```diff
@@ -64,6 +65,11 @@ folly::Future<Status> DescSpaceExecutor::execute() {
                 row.values.emplace_back(properties.get_charset_name());
                 row.values.emplace_back(properties.get_collate_name());
                 row.values.emplace_back(SchemaUtil::typeToString(properties.get_vid_type()));
+                if (properties.__isset.group_name) {
+                    row.values.emplace_back(properties.get_collate_name());
```
Comment (dutor) [inline src/executor/admin/SpaceExecutor.cpp:69]: Why to check `group_name` while accessing `collate_name`?

File: src/executor/admin/SpaceExecutor.cpp [hunk 3]
```diff
@@ -144,15 +150,17 @@ folly::Future<Status> ShowCreateSpaceExecutor::execute() {
...
                     fmt,
                     properties.get_space_name().c_str(),
                     properties.get_partition_num(),
                     properties.get_replica_factor(),
                     properties.get_charset_name().c_str(),
                     properties.get_collate_name().c_str(),
-                    SchemaUtil::typeToString(properties.get_vid_type()).c_str()));
+                    SchemaUtil::typeToString(properties.get_vid_type()).c_str(),
+                    properties.__isset.group_name ? properties.get_group_name()
+                                                  : new std::string("default")));
```
Comment (dutor) [inline src/executor/admin/SpaceExecutor.cpp:163]: It seems that you are trying to format a `std::string` like a C string? i.e. `%s`

File: src/parser/AdminSentences.h [hunk 4]
```diff
@@ -248,6 +249,15 @@ class SpaceOptItem final {
         }
     }
 
+    std::string getGroupName() const {
+        if (isString()) {
+            return asString();
+        } else {
+            LOG(ERROR) << "group name value illage.";
+            return 0;
```
Comment (dutor) [inline src/parser/AdminSentences.h:257]: Does this compile?

Comment (darionyaphet) [reply src/parser/AdminSentences.h:257]: it seems ok `getCharset` `getCollate` dose it with same way

File: src/executor/admin/SpaceExecutor.cpp (hunk 2 above)
Comment (darionyaphet) [reply src/executor/admin/SpaceExecutor.cpp:69]: Good catch

File: src/executor/admin/SpaceExecutor.cpp (hunk 3 above)
Comment (darionyaphet) [reply src/executor/admin/SpaceExecutor.cpp:163]: Setting the default value with `default` and it seems should be a `std::string*`

Comment (yixinglu) [reply src/executor/admin/SpaceExecutor.cpp:163]: Why not use `"default"` directly? It seems a memory leak.

File: src/parser/AdminSentences.h [hunk 5]
```diff
@@ -305,6 +315,11 @@ class CreateSpaceSentence final : public CreateSentence {
         spaceOpts_.reset(spaceOpts);
     }
 
+    void setGroupName(std::string* name) {
+        auto *item = new SpaceOptItem(SpaceOptItem::OptionType::GROUP_NAME, *name);
+        spaceOpts_->addOpt(std::move(item));
```
Comment (yixinglu) [inline src/parser/AdminSentences.h:320]: Why move a pointer?

File: src/parser/AdminSentences.h (hunk 4 above)
Comment (dutor) [reply src/parser/AdminSentences.h]: But it will crash.

File: tests/admin/test_space.py [hunk 6]
```diff
@@ -156,7 +156,7 @@ def test_charset_collate(self):
     def test_if_not_exists_and_if_exist(self):
         # exist then failed
         resp = self.client.execute('CREATE SPACE default_space')
-        self.check_resp_failed(resp)
+        self.check_resp_succeeded(resp)
```
Comment (CPWstatic) [inline tests/admin/test_space.py:159]: Why success?

**Review by CPWstatic — COMMENTED** [review]
We need also fix the `SHOW CREATE` test.

**Review by CPWstatic — APPROVED** [review]

File: src/parser/AdminSentences.h [hunk 7]
```diff
@@ -305,6 +315,11 @@ class CreateSpaceSentence final : public CreateSentence {
         spaceOpts_.reset(spaceOpts);
     }
 
+    void setGroupName(std::string* name) {
```
Comment (CPWstatic) [inline src/parser/AdminSentences.h:318]:
```
void setGroupName(std::string& name)
```

File: src/parser/parser.yy [hunk 8]
```diff
@@ -2791,11 +2791,23 @@ create_space_sentence
         auto sentence = new CreateSpaceSentence($4, $3);
         $$ = sentence;
     }
+    | KW_CREATE KW_SPACE opt_if_not_exists name_label KW_ON name_label {
+        auto sentence = new CreateSpaceSentence($4, $3);
+        sentence->setOpts(new SpaceOptList());
+        sentence->setGroupName($6);
+        $$ = sentence;
```
Comment (CPWstatic) [inline src/parser/parser.yy:2798]:
```
        sentence->setGroupName(*$6);
        $$ = sentence;
        delete $6;
```

**Review by yixinglu — APPROVED** [review]
Maybe we need some group tests in our CI system.

**Review by liuyu85cn — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**